test/test_dtype_alu.py

```diff
@@ -143,9 +143,10 @@ def test_bool(self, a, b, op): universal_test(a, b, dtypes.bool, op)
   @given(ht.int32, ht.int32, ht.float32, strat.sampled_from(integer_binary_operations), strat.sampled_from(binary_operations))
   def test_int32_midcast_float(self, a, b, c, op1, op2): universal_test_midcast(a, b, c, op1, op2, dtypes.int32, dtypes.float32)
 
-  # Metal and CUDACPU behave differently than numpy in CI for overflows
-  @given(strat.floats(width=32, min_value=0, max_value=10.0) if CI and (Device.DEFAULT == "METAL" or getenv("CUDACPU")) else ht.float32,
-         strat.floats(width=32, min_value=0, max_value=10.0) if CI and (Device.DEFAULT == "METAL" or getenv("CUDACPU")) else ht.float32,
+  # Metal and CUDACPU and HIP behave differently than numpy in CI for overflows
```

Comment:
This test fails on real HIP with the same error. I think it's something HIP-specific with how they generate the assembly.
![image](https://github.com/tinygrad/tinygrad/assets/77887910/0b4b750d-d4f0-4570-a690-9d15ca2ef1e1)